### PR TITLE
cp-29: Authenticated user flow

### PR DESCRIPTION
**Pull request by HIMDanny**
- Created a method for getting current user in the auth service
- Extended handler options with the authorized user object
- Created an endpoint for getting the current user in the auth controller

### CODE REVIEW

File: backend/src/libs/packages/controller/libs/types/api-handler-options.type.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ type DefaultApiHandlerOptions = {
   body?: unknown;
   query?: unknown;
   params?: unknown;
+  user?: unknown;
```
Comment (v0ldemar01) [inline backend/src/libs/packages/controller/libs/types/api-handler-options.type.ts]: let's provide type for it, auth user can have only single one

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: backend/src/packages/auth/auth.controller.ts [hunk 2]
```diff
@@ -33,6 +36,17 @@ class AuthController extends BaseController {
           }>,
         ),
     });
+
```
Comment (liza-veis) [inline backend/src/packages/auth/auth.controller.ts:39]: Add comments for swagger documentation as it's done for user service

File: backend/src/packages/auth/auth.controller.ts [hunk 3]
```diff
@@ -75,6 +89,15 @@ class AuthController extends BaseController {
       payload: await this.authService.signUp(options.body),
     };
   }
+
+  private async getUser(
```
Comment (liza-veis) [inline backend/src/packages/auth/auth.controller.ts]:
```suggestion
  private async getAuthenticatedUser(
```

File: backend/src/packages/auth/auth.service.ts [hunk 4]
```diff
@@ -16,6 +17,10 @@ class AuthService {
   ): Promise<UserSignUpResponseDto> {
     return this.userService.create(userRequestDto);
   }
+
+  public getUser(id: number): Promise<UserAuthResponseDto | null> {
```
Comment (liza-veis) [inline backend/src/packages/auth/auth.service.ts]:
```suggestion
  public getAuthenticatedUser(id: number): Promise<UserAuthResponseDto | null> {
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: backend/src/packages/auth/auth.controller.ts (hunk 2 above)
Comment (HIMDanny) [reply backend/src/packages/auth/auth.controller.ts:39]: done. Please tell me if something should be changed

File: backend/src/libs/packages/controller/libs/types/api-handler-options.type.ts [hunk 5]
```diff
@@ -1,7 +1,10 @@
+import { type UserAuthResponseDto } from '#packages/users/users.js';
+
 type DefaultApiHandlerOptions = {
   body?: unknown;
   query?: unknown;
   params?: unknown;
+  user?: UserAuthResponseDto | null;
```
Comment (v0ldemar01) [inline backend/src/libs/packages/controller/libs/types/api-handler-options.type.ts]: Not sure that user will have null value

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: backend/src/libs/packages/controller/libs/types/api-handler-options.type.ts (hunk 5 above)
Comment (HIMDanny) [reply backend/src/libs/packages/controller/libs/types/api-handler-options.type.ts]: so then I should change a user param to optional in the fastify request as well

Comment (v0ldemar01) [reply backend/src/libs/packages/controller/libs/types/api-handler-options.type.ts]: Good point, got it

**Review by v0ldemar01 — APPROVED** [review]

**Review by fshabanov — APPROVED** [review]

**Review by liza-veis — APPROVED** [review]

**Pull request merged**